filament/src/PostProcessManager.cpp

```diff
@@ -1467,10 +1480,56 @@ FrameGraphId<FrameGraphTexture> PostProcessManager::bloomPass(FrameGraph& fg,
                         driver.setMinMaxLevels(hwOut, i,
                                 i); // safe because we're using LINEAR_MIPMAP_NEAREST
                     }
+                    driver.setMinMaxLevels(hwOut, 0, bloomOptions.levels - 1);
                 });
 
         input = bloomDownsamplePass->out;
 
+        // flare pass
+        auto& flarePass = fg.addPass<BloomPassData>("Flare",
+                [&](FrameGraph::Builder& builder, auto& data) {
+                    data.in = builder.sample(input);
+                    data.out = builder.createTexture("Flare Texture", {
+                            .width  = width  / 2,
+                            .height = height / 2,
+                            .format = outFormat
+                    });
+                    data.out = builder.declareRenderPass(data.out);
+                },
+                [=](FrameGraphResources const& resources,
+                        auto const& data, DriverApi& driver) {
+                    auto in = resources.getTexture(data.in);
+                    auto out = resources.getRenderPassInfo(0);
+                    const float aspectRatio = float(width) / height;
+
+                    auto const& material = getPostProcessMaterial("flare");
+                    FMaterialInstance* mi = material.getMaterialInstance();
+
+                    mi->setParameter("color", in, {
+                            .filterMag = SamplerMagFilter::LINEAR,
+                            .filterMin = SamplerMinFilter::LINEAR_MIPMAP_NEAREST
+                    });
+
+                    mi->setParameter("level", 1.0f);    // adjust with resolution
```

Comment:
Is a todo needed here? Or an option?

Reply:
I think 1/16 res is a good choice for now (1/4 of the bloom buffer, which is already 1/4)